GUIDELINES.md

```diff
@@ -200,3 +227,41 @@ cluttering the output in the Travis-CI build environment.
 If an older version is required, put it in a subdirectory of the recipe.
 Examples of this can be found in [bowtie2](recipes/bowtie2),
 [bx-python](recipes/bx-python), and others.
+
+## ZLIB
+
+If a package depends on zlib, then it will most likely also depend on `gcc` (on
+Linux) and `llvm` (on OSX). The `meta.yaml` requirements section should
+therefore at least have the following:
+
+
+```yaml
+requirements:
+  build:
+    - gcc   # [not osx]
+    - llvm  # [osx]
+    - zlib
+
+  run:
+    - libgcc    # [not osx]
+    - zlib
+```
+
+When building the package, you may get an error saying that zlib.h can't be
+found -- despite having zlib listed in the dependencies. This is becuase the
+location of `zlib` often has to be specified in the `build.sh` script, which
+can be done like this:
+
+```bash
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
```

Comment:
In many cases `export C_INCLUDE_PATH=$PREFIX/include` is sufficient.  If this gets overwritten in some Makefile, `CFLAGS="-I$PREFIX/include"` is equivalent. But his one also get purged `CFLAGS=` in many Makefiles. LDFLAGS and CPPFLAGS are not so frequently used to fix these issue as far as I know.